core/blockly.js

```diff
@@ -274,17 +255,10 @@ Blockly.paste = function() {
  * @package
  */
 Blockly.duplicate = function(toDuplicate) {
-  // Save the clipboard.
-  var clipboardXml = Blockly.clipboardXml_;
-  var clipboardSource = Blockly.clipboardSource_;
-
-  // Create a duplicate via a copy/paste operation.
+  var data = Blockly.clipboardData_;
   Blockly.copy(toDuplicate);
-  toDuplicate.workspace.paste(Blockly.clipboardXml_);
-
-  // Restore the clipboard.
-  Blockly.clipboardXml_ = clipboardXml;
-  Blockly.clipboardSource_ = clipboardSource;
+  toDuplicate.workspace.paste(Blockly.clipboardData_.saveInfo);
+  Blockly.clipboardData_ = data;
```

Comment:
The above `Blockly.copy` call reassigns `clipboardData_` so we have to bring it back.

I guess you could change it to just `toDuplicate.workspace.paste(toDuplicate.toCopydata())`

I don't think it's worth it to change this since since the [new implementation](https://github.com/google/blockly/pull/5237/files#diff-96b05a61c1487616cf85c353d633e196e2aa643a7d00a7c0b2ca1877652d79a0R72) works.